.github/workflows/sonar.yml

```
@@ -0,0 +1,127 @@
+            });
+            var matchArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "target"
+            })[0];
+            var download = await github.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            var fs = require('fs');
+            fs.writeFileSync('${{github.workspace}}/target.zip', Buffer.from(download.data));
+      ## Extract the `target` directory from the build job
+      - name: Extract Build Result
+        run: |
+          unzip target.zip
+      ## Load the context from the build job - runs for any trigger to allow templates with `steps.build_context.outputs.content`
+      ## to be accepted by GitHub Actions.
+      - name: Read Build Context
+        id: build_context
+        uses: juliangruber/read-file-action@v1
+        with:
+          path: ./target/build-context.json
+
+      ## (PRs Only) Check out the base branch (target of the PR)
+      - name: Checkout Base Branch (PR Only)
+        if: github.event.workflow_run.event == 'pull_request'
+        env:
+          BASE_BRANCH: ${{ fromJson(steps.build_context.outputs.content).base_ref }}
+        run: |
+          git remote add upstream ${{ github.event.repository.clone_url }}
+          git fetch upstream --prune --tags --force
+          git checkout -B $BASE_BRANCH upstream/$BASE_BRANCH
+          git checkout ${{ github.event.workflow_run.head_sha }}
+          git clean -ffdx --exclude=target/ && git reset --hard HEAD
+      - name: Setup JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
+      - name: Cache Maven packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      ## (PRs Only) Run Sonar analysis against the results of the build job, providing PR information
+      - name: SonarCloud Analysis (PR Only)
+        if: github.event.workflow_run.event == 'pull_request'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{secrets.SONAR_TOKEN}}
+        run: |
+          mvn -B --no-transfer-progress org.sonarsource.scanner.maven:sonar-maven-plugin:sonar \
+            -Dsonar.projectKey=bf2fc6cc711aee1a0c2a_kas-fleetshard  \
+            -Dsonar.login=${SONAR_TOKEN} \
+            -Dsonar.coverage.jacoco.xmlReportPaths=$(pwd)/operator/target/site/jacoco-aggregate/jacoco.xml,$(pwd)/operator/target/site/jacoco/jacoco.xml,$(pwd)/sync/target/site/jacoco/jacoco.xml \
```

I think this and the push step below need to include the `api` and `common` coverage information and no aggregated report.  We should also remove the `<basepom.check.skip-coverage>` property from the `api` module.
```suggestion
            -Dsonar.coverage.jacoco.xmlReportPaths=$(pwd)/api/target/site/jacoco/jacoco.xml,$(pwd)/common/target/site/jacoco/jacoco.xml,$(pwd)/operator/target/site/jacoco/jacoco.xml,$(pwd)/sync/target/site/jacoco/jacoco.xml \
```

---

So, the XML files in the API module were empty because of `<basepom.check.skip-coverage>`. I tried removing the skip option, and the XML files have the results. I'll update this command.